docs/pages/bare/using-expo-client.md

```diff
@@ -53,121 +73,67 @@ if (Constants.appOwnership === 'expo') {
+</SnackInline>
+
+If you run this code in the Expo Go app you'll see the fallback view, if you were to build this locally with the [Expo CLI run commands](/workflow/expo-cli#compiling) then you'd see the native blur hash view.
 
-<ImageSpotlight alt="MapView working in an app built with Xcode and falling back to a placeholder in Expo Go" src="/static/images/expo-go-fallback.png" />
+## Native module detection
 
-As you may have already guessed, you can apply this concept for more than just third party dependencies. For example, maybe you wrote a native module to wrap your favorite analytics library and you want to provide a mock for it within the client.
+Native modules are added to the JavaScript global object by the runtime, this means you can conditionally check if they exist to ensure functionality:
 
 ```js
-// MyAnalytics.js
 import { NativeModules } from 'react-native';
 
-// Generic way of logging calls to MyAnalytics when module isn't available
-const makeLogger = name => options =>
-  console.log(`Called MyAnalytics.${name} with: ${JSON.stringify(options)}`);
+const isAvailable = !!NativeModules.MyAnalytics;
+```
 
-// Get native module or use fallback logger
-const MyAnalytics = NativeModules.MyAnalytics ?? {
-  logEvent: makeLogger('logEvent'),
-  setUser: makeLogger('setUser'),
-};
+This is nice because it ensures the native module _must_ be installed and linked. Two issues with this solution:
 
-// You usually want to wrap native function calls in functions on the JS side to
-// provide TypeScript typing and validation.
+1. You need to know the native module name ahead of time.
+2. You likely want an error to be thrown when a native module is missing in a custom build, this helps you determine if there was a native linking issue.
 
-export function logEvent(options) {
-  MyAnalytics.logEvent(options);
-}
+## Optional imports
 
-export function setUser(options) {
-  MyAnalytics.setUser(options);
-}
-```
-
-### Alternatively, use optional imports
-
-An alternative approach to using `expo-constants` as described above in the [conditional inline requires section](#use-conditional-inline-requires-to-provide-fallbacks) is to use `try/catch` around `require`. There's not any particularly good reason for you to use this in your application code, and it may lead to warning messages when the library throws an error on importing, but it is listed here in case you find yourself with a fitting use case.
+[Metro bundler](/guides/customizing-metro) has built-in support for optional imports. Optional imports refer to wrapping a `require` statement with a `try/catch` to prevent an error from being thrown when the requested module is missing:
```

Comment:
```suggestion
Optional imports are supported by [Metro bundler](/guides/customizing-metro). They refer to wrapping a `require` statement with a `try/catch` to prevent an error from being thrown when the requested module is missing:
```